rasa/shared/core/events.py

```diff
@@ -507,17 +585,170 @@ def create_external(
         )
 
 
+# noinspection PyProtectedMember
+class DefinePrevUserUttered(Event):
+    """Defines the family of events that are used to update previous user utterance."""
```

Comment:
what is a family of events and what do we need these events for? why does this need to be stored on the tracker and can not be recovered on the fly?

Reply:
do you suggest to run the prediction for all previous dialogue turns? I don't think it is the most efficient way. We also need to store entities predicted by TEDPolicy